server/server.js

```diff
@@ -1,30 +1,29 @@
 const express = require('express');
 const app = express();
-const port = 4000;
+const port = 4001;
+const cors = require('cors');
 const mongoose = require('mongoose');
 const db = require('../database/database.js');
 
-app.use('/:id', express.static(__dirname + '/../client/dist'));
+app.use(cors());
+app.use('/', express.static(__dirname + '/../public'));
+app.use('/:id', express.static(__dirname + '/../public'));
 
-// open mongoose connection
-mongoose.connect('mongodb://localhost/fec_product_features', {
-  useNewUrlParser: true,
-  useUnifiedTopology: true,
-  useFindAndModify: false,
-  useCreateIndex: true
-});
-
-mongoose.connection.on('error', console.error.bind(console, 'connection error:'));
+const corsOptions = {
+  origin: 'http://localhost:3000',
+  optionsSuccessStatus: 200
+}
```

Comment:
Did you run into any issue when you didn't have this piece of code? You will have more control if you make 'http://localhost' configurable instead of hardcoding.

Reply:
Yes I did! This code was added to be compatible with the proxy server that is run on port 3000.